client/my-sites/checkout/src/components/coupon.tsx

```diff
@@ -45,19 +45,38 @@ export default function Coupon( {
 				handleCouponSubmit();
 			} }
 		>
-			<Field
-				id={ id }
-				inputClassName="coupon-code"
-				value={ couponFieldValue }
-				disabled={ disabled || isPending }
-				placeholder={ String( translate( 'Enter your coupon code' ) ) }
-				isError={ hasCouponError && ! isFreshOrEdited }
-				errorMessage={ errorMessage }
-				onChange={ ( input ) => {
-					setIsFreshOrEdited( true );
-					setCouponFieldValue( input );
-				} }
-			/>
+			{ hasCheckoutVersion( '2' ) ? (
+				<>
```

Comment:
It's minor but you could wrap just one version of this field in a fragment and then conditionally show the `CouponLabel` for v2 checkout. That way you wouldn't need to repeat this whole block. I think it should be safe because `CouponField` just sets a CSS grid variable which should be ignored for v1 checkout.

Reply:
Hmm, that's a good idea but I think I'm not too worried about conciseness here since we'll be removing the old version soon anyway.